Azure/azure-cli
# [Keyvault] `az keyvault key sign/verify`: Fix `--digest` to accept base64 encoded string

**Related command**

`az keyvault key sign/verify`

**Description**

Fix https://github.com/Azure/azure-cli/issues/27631, https://github.com/Azure/azure-cli/issues/28027

`--digest` now accepts base64 encoded string

**Testing Guide**

```
> echo "HelloWorld" | openssl dgst -sha256 -binary | openssl base64
Pz8hPx8/Pz9mP11RFj8/Mz8/bj91PwRhPz8JQ2k/eGMNCg==
> az keyvault key sign -n $rsaKey --vault-name $testVault -a RS256 --digest Pz8hPx8/Pz9mP11RFj8/Mz8/bj91PwRhP
{
  "algorithm": "RS256",
  "keyId": "https://$testVault.vault.azure.net/keys/......",
  "signature": "UZmopfF5fHtGs+qu8............=="
}
> az keyvault key verify -n $rsaKey --vault-name $testVault -a RS256 --digest Pz8hPx8/Pz9mP11RFj8/Mz8/bj91PwRhP --signature "UZmopfF5fHtGs+qu8............=="
{
  "algorithm": "RS256",
  "isValid": true,
  "keyId": "https://$testVault.vault.azure.net/keys/......"
}
> az keyvault key sign -n $ecKey --vault-name $testVault -a ES256 --digest Pz8hPx8/Pz9mP11RFj8/Mz8/bj91PwRhP
> az keyvault key verify -n $ecKey --vault-name $testVault -a ES256 --digest Pz8hPx8/Pz9mP11RFj8/Mz8 --signature "......"
```

**History Notes**


[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.
